File: app/src/main/java/cz/kutner/comicsdb/holder/ComicsViewHolder.java
package cz.kutner.comicsdb.holder;

import android.content.Intent;
import android.view.View;
import android.widget.TextView;

import butterknife.ButterKnife;
import butterknife.OnClick;
import cz.kutner.comicsdb.R;
import cz.kutner.comicsdb.activity.ComicsDetailActivity;
import cz.kutner.comicsdb.model.Comics;
import uk.co.ribot.easyadapter.ItemViewHolder;
import uk.co.ribot.easyadapter.PositionInfo;
import uk.co.ribot.easyadapter.annotations.LayoutId;
import uk.co.ribot.easyadapter.annotations.ViewId;

@LayoutId(R.layout.list_item_comics)
public class ComicsViewHolder extends ItemViewHolder<Comics> {
    public final static String COMICS_ID = "cz.kutner.comicsdbclient.comicsdbclient.comics_id";

    @ViewId(R.id.comics_name)
    TextView comicsName;
    @ViewId(R.id.comics_published)
    TextView comicsPublished;
    @ViewId(R.id.comics_rating)
    TextView comicsRating;
    Integer comicsId;

    public ComicsViewHolder(View view) {
        super(view);
        ButterKnife.bind(this, view);
    }

    @OnClick(R.id.card_view_comics)
    public void onComicsClick(View v) {
        Intent intent = new Intent(v.getContext(), ComicsDetailActivity.class);
        intent.putExtra(COMICS_ID, comicsId);
        v.getContext().startActivity(intent);
    }

    @Override
    public void onSetValues(Comics comics, PositionInfo positionInfo) {
        comicsName.setText(comics.getName());
        comicsPublished.setText(comics.getPublished());
        if (comics.getRating() > 0) {
            comicsRating.setText(comics.getRating().toString());
        } else {
            comicsRating.setText(" ");
        }
        comicsId = comics.getId();
    }
}

File: app/src/main/kotlin/cz/kutner/comicsdb/holder/ComicsViewHolder.kt
package cz.kutner.comicsdb.holder

import android.content.Intent
import android.support.v7.widget.CardView
import android.view.View
import android.widget.TextView
import cz.kutner.comicsdb.R
import cz.kutner.comicsdb.activity.ComicsDetailActivity
import cz.kutner.comicsdb.activity.MainActivity
import cz.kutner.comicsdb.model.Comics
import org.jetbrains.anko.onClick
import uk.co.ribot.easyadapter.ItemViewHolder
import uk.co.ribot.easyadapter.PositionInfo
import uk.co.ribot.easyadapter.annotations.LayoutId
import uk.co.ribot.easyadapter.annotations.ViewId

@LayoutId(R.layout.list_item_comics)
class ComicsViewHolder(view: View) : ItemViewHolder<Comics>(view) {

    @ViewId(R.id.comics_name)
    internal lateinit var comicsName: TextView
    @ViewId(R.id.comics_published)
    internal lateinit var comicsPublished: TextView
    @ViewId(R.id.comics_rating)
    internal lateinit var comicsRating: TextView
    internal var comicsId: Int? = null
    @ViewId(R.id.card_view_comics)
    internal lateinit var card_view_comics: CardView

    init {
        card_view_comics.onClick {
            val intent = Intent(view.context, ComicsDetailActivity::class.java)
            if (comicsId != null) {
                intent.putExtra(MainActivity.COMICS_ID, comicsId as Int)
                view.context.startActivity(intent)
            }
        }
    }

    override fun onSetValues(comics: Comics, positionInfo: PositionInfo) {
        comicsName.text = comics.name
        comicsPublished.text = comics.published
        if (comics.rating.toInt() > 0) {
            comicsRating.text = comics.rating.toString()
        } else {
            comicsRating.text = " "
        }
        comicsId = comics.id
    }
}
